gcloud/resource_manager/client.py

```diff
@@ -0,0 +1,238 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""A Client for interacting with the Resource Manager API.
+
+Overview
+~~~~~~~~
+
+There are three main methods in the ``Client`` class:
+
+- :func:`gcloud.resource_manager.client.Client.list_projects`
+- :func:`gcloud.resource_manager.client.Client.project`
+- :func:`gcloud.resource_manager.client.Client.get_project`
+
+``project()`` versus ``get_project()``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The difference between ``project`` and ``get_project`` is subtle,
+so it might be worthwhile to make a quick distinction.
+
+If you want to simply "get a hold of a Project" object,
+but **don't** want to actually retrieve any metadata about that project
+(for example, when you want to create a new project, or delete a project
+in a single API request), ``project()`` is the best method to use::
+
+    >>> from gcloud import resource_manager
+    >>> client = resource_manager.Client()
+    >>> project = client.project('purple-spaceship-123')
+    >>> project.number is None
+    True
+
+The ``project`` referenced above has no extra metadata associated with it,
+however you can still operate on it (ie, ``project.create()`` or
+``project.delete()``).
+
+If you want to retrieve a project and all of it's metadata, the best method
+to use is ``get_project()``, which will return ``None`` if the project
+doesn't exist::
+
+    >>> from gcloud import resource_manager
+    >>> client = resource_manager.Client()
+    >>> project = client.get_project('purple-spaceship-123')
+    >>> project.number is None
+    False
+    >>> project = client.get_project('doesnt-exist')
+    >>> project is None
+    True
+"""
+
+
+from gcloud.client import Client as BaseClient
+from gcloud.exceptions import NotFound
+from gcloud.iterator import Iterator
+from gcloud.resource_manager.connection import Connection
+from gcloud.resource_manager.project import Project
+
+
+class Client(BaseClient):
+    """Client to bundle configuration needed for API requests.
+
+    See
+    https://cloud.google.com/resource-manager/reference/rest/
+    for more information on this API.
+
+    Automatically get credentials::
+
+        >>> from gcloud.resource_manager import Client
+        >>> client = Client()
+
+    .. note::
+
+        Chances are you want to use either the constructor with no arguments,
+        or one of the factory methods (like
+        :func:`gcloud.resource_manager.client.Client.from_service_account_json`
+        or similar).
```

Comment:
We don't want the factory constructors to be used, since only a user can authenticate to manage projects, so this note should be squashed. You may even want to consider overwriting the methods from the superclass:

```python
    @classmethod
    def from_service_account_json(cls, *args, **kwargs):
        raise NotImplementedError(args, kwargs)

    @classmethod
    def from_service_account_p12(cls, *args, **kwargs):
        raise NotImplementedError(args, kwargs)
```